syntax = "proto3";
package bentoml;

message LabelSelectors {
    message LabelSelectorExpression {
        enum OPERATOR_TYPE {
            In = 0;
            NotIn = 1;
            Exists = 2;
            DoesNotExist = 3;
        }
        OPERATOR_TYPE operator = 1;
        string key = 2;
        repeated string values = 3;
    }
    map<string, string> match_labels = 1;
    repeated LabelSelectorExpression match_expressions = 2;
}
